runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -504,7 +510,9 @@ private void updateList()
 		WorldResult worldResult = worldService.getWorlds();
 		if (worldResult != null)
 		{
-			SwingUtilities.invokeLater(() -> panel.populate(worldResult.getWorlds()));
+			SwingUtilities.invokeLater(() -> panel.populate(
+				worldResult.getWorlds(),
+				getIsWestCoastWorld()));
```

Comment:
the panel has a back reference to the plugin (which is even used in populate), it is probably better just to have it call a method there.

Reply:
Changed to set a new property on the panel with the west coast worlds from the client thread before panel.populate() is called. SwingUtilities.invokeLater is now triggered via the client thread, then eventually the "AWT-EventQueue" thread actually runs panel.populate() with all the required world information loaded. There is no race condition failure since if for whatever reason the client thread takes forever to run before panel.populate() runs, we will load the original US flag icon without showing East/West.